modules/client/src/client-common/context.ts

```
@@ -1,116 +1,151 @@
+      this.state.graphql = Context.resolveGraphql(contextParams.graphqlNodes);
+      this.manualState.graphql = true;
+    }
+    if (contextParams.ipfsNodes?.length) {
+      this.state.ipfs = Context.resolveIpfs(contextParams.ipfsNodes);
+      this.manualState.ipfs = true;
+    }
+    if (contextParams.daoFactoryAddress) {
+      this.state.daoFactoryAddress = contextParams.daoFactoryAddress;
+      this.manualState.daoFactory = true;
+    }
+    if (contextParams.ensRegistryAddress) {
+      this.state.ensRegistryAddress = contextParams.ensRegistryAddress;
+      this.manualState.ensRegistry = true;
     }
     if (contextParams.gasFeeEstimationFactor) {
       this.state.gasFeeEstimationFactor = Context.resolveGasFeeEstimationFactor(
-        contextParams.gasFeeEstimationFactor
+        contextParams.gasFeeEstimationFactor,
       );
+      this.manualState.gasFeeEstimationFactor = true;
     }
+  }
 
-    if (contextParams.ipfsNodes?.length) {
-      this.state.ipfs = Context.resolveIpfs(contextParams.ipfsNodes);
+  private setDefault() {
+    const networkName = this.state.network.name as SupportedNetworks;
+    if (
+      WEB3_NODES[networkName] && WEB3_NODES[networkName].length &&
+      !this.manualState.web3
+    ) {
+      this.state.web3Providers = Context.resolveWeb3Providers(
+        WEB3_NODES[networkName],
+        this.state.network,
+      );
     }
-    if (contextParams.graphqlNodes?.length) {
-      this.state.graphql = Context.resolveGraphql(contextParams.graphqlNodes);
+    if (
+      GRAPHQL_NODES[networkName] && GRAPHQL_NODES[networkName].length &&
+      !this.manualState.graphql
+    ) {
+      this.state.graphql = Context.resolveGraphql(GRAPHQL_NODES[networkName]);
+    }
+    if (
+      IPFS_NODES[networkName] && IPFS_NODES[networkName].length &&
+      !this.manualState.ipfs
+    ) {
+      this.state.ipfs = Context.resolveIpfs(IPFS_NODES[networkName]);
+    }
+    if (LIVE_CONTRACTS[networkName]) {
```

Same, about throwing an error instead of failing silently